wordpress-mobile/WordPress-Android
# [JP Content Migration Flow] Enables Jetpack migration flow

Fixes #17528

## Description
Enables the Jetpack migration flow by flipping on the following build flags:
* `JETPACK_POWERED_BOTTOM_SHEET `
* `JETPACK_SHARED_LOGIN `
* `JETPACK_LOCAL_USER_FLAGS `
* `JETPACK_BLOGGING_REMINDERS_SYNC `
* `JETPACK_READER_SAVED_POSTS `
* `JETPACK_PROVIDER_SYNC `
* `JETPACK_MIGRATION_FLOW `

## Notes:
* The `PREVENT_DUPLICATE_NOTIFS_REMOTE_FIELD` flag will be enabled with https://github.com/wordpress-mobile/WordPress-Android/pull/17513
* The `OPEN_WEB_LINKS_WITH_JETPACK_FLOW ` flag has not been enabled with this PR (cc @zwarm)
* Since [the minimum supported version check has been merged](https://github.com/wordpress-mobile/WordPress-Android/pull/17536/files#diff-4cc25132735e324ad2e5806549f25b9aac7835554e0a7998fab099fd75a8dd56R23) in order **to test the flow in debug you need to change this to something lower (e.g. 21.2)**

## To test:

<details>
<summary>1. Testing the Jetpack powered bottom sheet</summary>

1. Launch the WordPress app
2. **Verify** that the bottom sheet overlay is show when the user taps on a badge/banner in the following cases:
  * **Badges**:
     1. `Dashboard` → `Home` tab
     2. `Me`
     4. `Activity Log` → `Activity Item Detail`
     5. `Me` → `App Settings`
     6. `Notifications` → `Notifications Settings`
     7. `Sharing`
     8. `Reader` → `Reader Detail`
  * **Banners**
    1. `Reader` list
    2. `Reader` → `Search` (both empty and with search suggestions that show after searching once)
    3. `Reader` → `Search` → `Search results` list
    5. `Notifications` list
    6. `Stats` page
    7. `Activity Log` list

</details>

<details>
<summary>2. Generate some local-only post/page content in the WordPress App</summary>

1. Install the WordPress app flavor from this branch
2. Login to the WordPress app and select a site
3. Publish a post
4. Make changes to the post and do not publish them (local changes)
_You could create more as described in https://github.com/wordpress-mobile/WordPress-Android/pull/17398_

</details>

<details>
<summary>3. Create some local settings</summary>

1. Install the WordPress app flavor from this branch
2. Go to `Posts` list and tap the circle icon under the `←` back arrow, to filter posts by author
3. Select `Me` instead of `Everyone`
4. After fresh installing, running the Jetpack App and completing the migration flow (main steps 6-8):
   1. Go to Posts list
   2. **Verify** the posts filtering by author is set to `Me`

</details>

<details>
<summary>4. Create a blogging reminder</summary>

1. Install the WordPress app flavor from this branch
2. On `My Site` go to `Menu` → `Site Settings` → `Blogging Reminders`
3. Select at least a day for blogging reminders and tap `Update`
4. After fresh installing, running the Jetpack App and completing the migration flow (main steps 6-8):
   1. Go from `My Site` to `Menu` → `Site Settings` → `Reminders and prompts`
   2. **Verify** the blogging reminder set at step `2` exists in the Jetpack app (same day(s) selected)

</details>

<details>
<summary>5. Save some reader posts</summary>

1. Install the WordPress app flavor from this branch
2. Go to `Reader` → `Discover`
3. Tap the bookmark/save icon (first icon) under one post and continue on the dialog that appears
4. Go to `Saved` tab to make sure you have a saved post (note it down)
4. After fresh installing, running the Jetpack App and completing the migration flow (main steps 6-8):
   1. Go to `Reader`
      Note (Reader will open automatically if you left the WordPress app open on `Reader`)
   2. Go to `Saved` tab
   3. **Verify** The post saved at step 3 & verified at step 4 exists in the Jetpack app
</details>

 _Uninstall the Jetpack app or log out of Jetpack and delete all data._

6. Run the Jetpack app from this branch
7. Expect not see the migration flow
8. Proceed till the end
9. **Verify** that you logged in automatically and that your data are available

## Regression Notes
1. Potential unintended areas of impact
N/A

5. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

6. What automated tests I added (or what prevented me from doing so)
Tests where added during feature development

PR submission checklist:
